### PR TITLE
Structured json file

### PR DESCRIPTION
**Pull request by CanBerker**

### CODE REVIEW

File: src/main/java/sparksoniq/spark/iterator/function/StructuredJsonFileFunctionIterator.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    public StructuredJsonFileFunctionIterator(List<RuntimeIterator> arguments, IteratorMetadata iteratorMetadata) {
+        super(arguments, iteratorMetadata);
+    }
+
+    @Override
+    public void open(DynamicContext context) {
+        super.open(context);
+
+        long resultSize = this.getRDD(_currentDynamicContext).count();
+        this._hasNext = resultSize != 0;
```
Comment (ghislainfourny) [inline src/main/java/sparksoniq/spark/iterator/function/StructuredJsonFileFunctionIterator.java]: Why is this needed?

Comment (CanBerker) [reply src/main/java/sparksoniq/spark/iterator/function/StructuredJsonFileFunctionIterator.java]: Great catch! After taking a look at this I have noticed that this redundant override exists for not only structuredJSON file but all of RDDRuntimeIterator implementations. I have removed them as well and updated the PR.

File: src/main/resources/test_files/runtime-spark/JsonFile.jq [hunk 2]
```diff
@@ -1,2 +1,2 @@
 (:JIQS: ShouldRun; Output="({ "guess" : "Latvian", "target" : "Russian", "country" : "AU", "choices" : [ "Lao", "Latvian", "Russian", "Swahili" ], "sample" : "b7df3f9d67cef259fbcaa5abcad9d774", "date" : "2013-08-20" }, { "guess" : "Russian", "target" : "Russian", "country" : "AU", "choices" : [ "Croatian", "Nepali", "Russian", "Slovenian" ], "sample" : "8a59d48e99e8a1df7e366c4648095e27", "date" : "2013-08-20" }, { "guess" : "Czech", "target" : "Czech", "country" : "SE", "choices" : [ "Maori", "Czech", "Korean", "Turkish" ], "sample" : "1787b5c79a00b3513ce76847bc1f5b75", "date" : "2013-08-20" }, { "guess" : "Greek", "target" : "Serbian", "country" : "SE", "choices" : [ "German", "Greek", "Kannada", "Serbian" ], "sample" : "0d5b697ebb326b5043ce7fa60a7b968d", "date" : "2013-08-20" }, { "guess" : "Serbian", "target" : "Serbian", "country" : "AU", "choices" : [ "Dari", "Serbian", "Sinhalese", "Vietnamese" ], "sample" : "0d5b697ebb326b5043ce7fa60a7b968d", "date" : "2013-08-20" })" :)
-json-file("./src/main/resources/queries/conf-ex.json", 10)
\ No newline at end of file
+json-file("./src/main/resources/queries/conf-ex.json", 10)
```
Comment (ghislainfourny) [inline src/main/resources/test_files/runtime-spark/JsonFile.jq]: Note for later (the review reminded me of this): could we add, at some point (nothing urgent), tests on empty files, to check that nothing crashes and an empty sequence is, indeed, returned?

File: src/main/java/sparksoniq/spark/iterator/function/StructuredJsonFileFunctionIterator.java (hunk 1 above)
Comment (ghislainfourny) [reply src/main/java/sparksoniq/spark/iterator/function/StructuredJsonFileFunctionIterator.java]: Very nice job! Thanks.

**Comment by ghislainfourny** [comment]
@CanBerker One more thing: would it be possible to edit the function list in the documentation, and add structured-json-file() with a few words of explanation? Thanks!

File: src/main/resources/test_files/runtime-spark/JsonFile.jq (hunk 2 above)
Comment (CanBerker) [reply src/main/resources/test_files/runtime-spark/JsonFile.jq]: I have added the tests for reading the empty files. The parquet file does not have the emptyFile test however (it is disabled) as it doesn't fit into the parquet format. What kind of a test would you like to add there ?

**Comment by ghislainfourny** [comment]
@CanBerker yes, that's good to go -- please just remember the documentation at some point.

**Comment by CanBerker** [comment]
@ghislainfourny The documentation is updated. Does it look sufficient or should I add more details?

**Comment by ghislainfourny** [comment]
Thanks a lot! It looks good.

Let's put empty parquet files aside for now and worry about them later. I'll try to get empty samples for our tests.
